### PR TITLE
Feat/authentication metrics

### PR DESCRIPTION
**Pull request by SamMHD**

### CODE REVIEW

File: pkg/auth/metrics.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		prometheus.GaugeOpts{
+			Name: "cache_read_locks",
+			Help: "Number of active read_lock on the cache",
+		},
+	)
+
+	accessCacheEntries = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "access_cache_entries",
+			Help: "",
```
Comment (Cypherspark) [inline pkg/auth/metrics.go:80]: Why this metric and the following two other metrics has no help description?

File: pkg/auth/authenticator.go [hunk 2]
```diff
@@ -221,19 +293,27 @@ func (a *Authenticator) Check(ctx context.Context, request *Request) (*Response,
 		response.Header.Add(key, value)
 	}
 
+	// update metrics
+	reqCount.With(ReasonLabel(reason)).Inc()
+	reqLatency.With(ReasonLabel(reason)).Observe(time.Since(reqStartTime).Seconds())
```
Comment (Cypherspark) [inline pkg/auth/authenticator.go:298]: Shouldn't we calculate request time based on Check func inside server.go
([here](https://github.com/snapp-incubator/Cerberus/blob/2549ff38ba71b46f07d2c8f67cd10ca22e59d86f/pkg/auth/server.go#L33)) 
so we can include the conversion time too?

Comment (SamMHD) [reply pkg/auth/authenticator.go:298]: That's a good Idea. I'll change it

Comment (SamMHD) [reply pkg/auth/authenticator.go:298]: resolved in https://github.com/snapp-incubator/Cerberus/pull/13/

**Comment by SamMHD** [comment]
@Cypherspark, your suggestions added in https://github.com/snapp-incubator/Cerberus/pull/13/
Please review them 🙏

**Pull request merged**